### ISSUE TITLE
Using with mongodb_ecto results in invalid spans (the entire trace disappears from DataDog)

### ISSUE THREAD

**Issue by vovayartsev** [spandex-project/spandex_ecto#9]
Steps to reproduce:

1. Use Mongo.Ecto `      {:mongodb_ecto, github: "kkvesper/mongodb_ecto", branch: "ecto-2.2"},`
2. Integrate `spandex_ecto` as usual by addong `SpandexEcto.EctoLogger` to the repo's loggers 
3. Launch some REST Phoenix webapp which renders data from Mongo and watch APM traces in Datadog

Expected: 

1. Traces from Phoenix should be visible in Datadog
2. The traces should include spans from Ecto service (even if specific Mongo query can't be identified, it's still better than nothing)

Actual

1. Phoenix traces don't appear in Datadog anymore
2. In Datadog Agent there are errors like this:

```
[ TRACE ] 2018-12-14 01:10:01 ERROR (receiver.go:249) - dropping trace reason: invalid span (SpanID:6223878502410899477): span.normalize: empty `Resource` (debug for more info), [service:"ecto" name:"query"...
```

Thoughts:

"empty Resource" error happens when the function below returns an empty string.

```
  defp string_query(_), do: ""
```

It returns `""` because LogEntry for Ecto.Mongo looks like this:

```
%Ecto.LogEntry{
  ansi_color: nil,
  caller_pid: nil,
  connection_pid: nil,
  decode_time: 9000,
  params: [],
  query: [
    "FIND",
    [" ", "coll", "=" | "\"tickets_\""],
    [" ", "query", "=" | "[{\"$query\", [status: 4]}, {\"$orderby\", %{}}]"],
    [" ", "projection", "=" |
     "%{_id: true, bar_code: true, cost: true, event_id: true, fees_breakdown: true, flags: true, owner_id: true, purchased_at: true, qr_code: true, status: true}"]
  ],
  query_time: 253000,
  queue_time: 20000,
  result: {:ok, {:op_reply, 8, 0, 0, 0, []}},
  source: nil
}
```

Assuming that proper mongodb_ecto support is out of scope, just replacing `""` resource with `"unknown"` somewhere before sending to Datadog would solve the issue.

**Comment by GregMefford** [comment]
Thanks for reporting this. You're correct that we simply didn't have `mongodb_ecto` in mind for this plugin, but I think there should be an easy work-around for you: In the latest version of `spandex_datadog`, we have a bugfix that allows you to send spans that are missing the `resource` field, and it will just copy the `name` field as the `resource`, since we already require that `name` always be set. Please check your `mix.lock` file and see whether you are using `spandex_datadog` `v0.3.1`.

**Comment by vovayartsev** [comment]
My mix.lock already reads:
```
"spandex_datadog": {:hex, :spandex_datadog, "0.3.1", "984d27ad1f45cfd243509692f0f63b900a23b79566c529a644c7f3a2b4120603", ...
```

I believe you mean [this fix](https://github.com/spandex-project/spandex_datadog/commit/b2dd1c0b77dbc6eba60953efc2b5f2bcb8ab1a72), however it only fixes `nil` resources, while `defp string_query(_)` returns empty string `""`.

**Comment by zachdaniel** [comment]
@vovayartsev The fix on your fork looks great. If you'd like to PR it, I'll get it released!

**Comment by zachdaniel** [comment]
Although we *may* want to make that custom behavior, so you can say something like

```
config :spandex_ecto, SpandexEcto.EctoLogger,
  query_formatter: {Mod, :fun, []}
```

**Comment by vovayartsev** [comment]
Oh, I missed the response and ignored it for almost a year!

@zachdaniel Do you still prefer me to make the PR? Or just close the issue as an outdated one, now that the project is moving in `telemetry` direction?

**Comment by zachdaniel** [comment]
Might as well fix it, I’ll review your PR if it’s up, but I don’t mind either way. You can close the issue or PR the fix, your call :)

**Comment by vovayartsev** [comment]
Thanks. I rebased and created a PR: https://github.com/spandex-project/spandex_ecto/pull/18 hoping this helps somebody.
I didn't improve `EctoLogger` with `query_formatter` because I believe in `telemetry` approach.

**Comment by zachdaniel** [comment]
Fix released as `0.6.1`
